### PR TITLE
feat: load data using S3

### PR DESCRIPTION
**Pull request by dymbe**
Load flood data in-memory from S3 on startup and start a task that reloads the data at 12:00 UTC every day

~~Should consider logging in a more structured way, but this is true for most of our services~~ (done)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ license = "Apache-2.0"
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = "^3.11"
+python = "^3.11,<3.13"
```
Comment (dymbe) [inline pyproject.toml:10]: This was necessary to install Pandas

**Review by giltinde — APPROVED** [review]
Nice work 👍

File: flood_api/dependencies/flooddata.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+SummaryFloodDataDep = Annotated[gpd.GeoDataFrame, Depends(get_summary_data)]
+DetailedFloodDataDep = Annotated[gpd.GeoDataFrame, Depends(get_detailed_data)]
+
+
+async def flood_data_fetcher(app: FastAPI):
+    print("Flood data fetcher running")
+    schedule = crontab("0 12 * * * *", tz=timezone.utc)
+    while True:
```
Comment (kstigen) [inline flood_api/dependencies/flooddata.py:48]: How does this play with the fastapi/uvicorn shutdown? Is it possible to shut down the application gracefully?

File: flood_api/settings.py [hunk 3]
```diff
@@ -7,7 +7,13 @@ class Settings(BaseSettings):
     uvicorn_host: str = "0.0.0.0"
     uvicorn_reload: bool = True
     uvicorn_proxy_headers: bool = False
-    database_url: str = "postgresql+psycopg://postgres:postgres@localhost:5432/postgres"
+    flood_data_path_detailed: str = (
+        "s3://openepi-public-data/processed_detailed_forecast.parquet/"
```
Comment (kstigen) [inline flood_api/settings.py:11]: Should this also be overridden by an environment variable perhaps?

File: flood_api/routers/flood.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+from flood_api.dependencies.flooddata import DetailedFloodDataDep
+from flood_api.dependencies.queryparams import CoordinatesDep, DateRangeDep
+
+router = APIRouter(prefix="/flood", tags=["flood"])
+
+
+@router.get("/detailed")
+async def detailed(
+    data: DetailedFloodDataDep, coordinates: CoordinatesDep, date_range: DateRangeDep
```
Comment (kstigen) [inline flood_api/routers/flood.py:11]: Is it possible to also get documentation for these input types in addition to the data type?

File: flood_api/__main__.py [hunk 5]
```diff
@@ -1,11 +1,29 @@
...
+from flood_api.settings import settings
+
+import asyncio
+
 
+@asynccontextmanager
+async def lifespan(flood_app: FastAPI):
+    await fetch_flood_data(flood_app)
+    asyncio.create_task(flood_data_fetcher(flood_app))
+    yield
```
Comment (kstigen) [inline flood_api/__main__.py:16]: Should there be something after this yield in order to shut down the task of fetching new data?

File: flood_api/dependencies/flooddata.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        engine="fastparquet",
+        storage_options={"anon": True},
+    )
+
+    # Convert WKT strings to geometry objects
+    df["geometry"] = df["wkt"].apply(wkt.loads)
+
+    gdf = gpd.GeoDataFrame(df, geometry="geometry").drop(columns="wkt")
+
+    print(f"Done reloading data from {path}")
```
Comment (kstigen) [inline flood_api/dependencies/flooddata.py:28]: If this is meant to be logging, we should change this print to use the logging library.

**Review by kstigen — CHANGES_REQUESTED** [review]

File: flood_api/dependencies/flooddata.py (hunk 2 above)
Comment (dymbe) [reply flood_api/dependencies/flooddata.py:48]: It seems to be shutting down without any problems when the main-thread is shutting down

File: flood_api/__main__.py (hunk 5 above)
Comment (dymbe) [reply flood_api/__main__.py:16]: The flood_data_fetcher-task seems to be shutting down without any problems when the main-thread is shutting down. I can't think of anything else that needs to be cleaned up

File: flood_api/dependencies/flooddata.py (hunk 6 above)
Comment (dymbe) [reply flood_api/dependencies/flooddata.py:28]: Agreed

File: flood_api/settings.py (hunk 3 above)
Comment (dymbe) [reply flood_api/settings.py:11]: I was considering that too, but I thought this was a convenient default for the moment.

Would you want me to hardcode the S3-paths in the k8s-YAML instead? I'm not sure if I see what we gain by moving it there.

A better option would perhaps be to deploy the data-bucket in IaC, and then create a ConfigMap that could be referenced in the k8s-YAML. The fact that we only have one databricks environment, but several environments for the rest of the platform makes this a bit more complicated though

**Review by kstigen — APPROVED** [review]

**Review by kstigen — APPROVED** [review]
LGTM

**Review by giltinde — APPROVED** [review]

**Pull request merged**